examples/random_forest_example.py

```diff
@@ -0,0 +1,169 @@
+import shapely
+
+from openeo.processes import ProcessBuilder, mean, array_create
+from shapely.geometry import Polygon
+import openeo
+from openeo.udf import FeatureCollection
+from geopandas import GeoDataFrame
+
+polygon1 = Polygon([(16.138916, 48.1386), (16.138916, 48.320647), (16.2, 48.320647), (16.2, 48.1386)])
+polygon2 = Polygon([(16.3, 48.1386), (16.3, 48.320647), (16.5, 48.320647), (16.5, 48.1386)])
+polygons = shapely.geometry.GeometryCollection([polygon1, polygon2])
+
+con = openeo.connect("https://openeocloud-dev.vito.be/openeo/1.0")
+
+## Features ##
+temp_ext_s2 = ["2020-04-02", "2020-04-20"]
+spat_ext_s2 = {
+    'west': 16.138916, 'east': 16.524124,
+    'south': 48.1386, 'north': 48.320647
+    }
+cube = con.load_collection("boa_landsat_8",
+                           spatial_extent=spat_ext_s2,
+                           temporal_extent=temp_ext_s2,
+                           bands=["B03", "B04"])
+S2_month = cube.aggregate_temporal_period(period="month", reducer="mean")
+s2_without_nodata = S2_month.apply_dimension(dimension="t", process="array_interpolate_linear")
+
+
+def compute_stats(input_timeseries: ProcessBuilder):
+    # Only take the first 5 months.
+    tsteps = list([input_timeseries.array_element(index) for index in range(0, 5)])
+    return array_create(tsteps)
+
+
+s2_time_flattened = s2_without_nodata.apply_dimension(dimension='t', target_dimension='bands', process=compute_stats)
+tstep_labels = ["t" + str(index) for index in range(0, 5)]
+all_bands = [band + "_" + stat for band in cube.metadata.band_names for stat in tstep_labels]
+s2_time_flattened = s2_time_flattened.rename_labels('bands', all_bands)
+
+# 1 pixel per polygon with #bands features.
+# Later features will be points that are randomly sampled from the original polygons.
+features = s2_time_flattened.aggregate_spatial(geometries=polygons, reducer=mean)
+
+## Target ##
+labels = FeatureCollection(
```

Comment:
as discussed elsewhere: `openeo.udf.FeatureCollection`  is not the right feature collection wrapper to be used here